plugins/maciVoting/services/coordinator/coordinator.ts

```diff
@@ -0,0 +1,175 @@
+import { type Address } from "viem";
+import { type z } from "zod";
+import { ESupportedNetworks } from "./utils/networks";
+import { type IGenerateArgs, type ISubmitProofsArgs } from "./utils/types";
+import { getAuthorizationHeader } from "./utils/auth";
+import { getSigner, getPublicClient } from "./utils/chain";
+import { pollDeploymentConfig } from "./config";
+import { GenerateResponseSchema, SubmitResponseSchema } from "./utils/schemas";
+
+const BASE_URL = "http://localhost:3000/v1";
+const CHAIN = ESupportedNetworks.ETHEREUM_SEPOLIA;
+
+type GenerateResponse = z.infer<typeof GenerateResponseSchema>;
+type SubmitResponse = z.infer<typeof SubmitResponseSchema>;
+type CoordinatorServiceResult<T, E = Error> = { success: true; data: T } | { success: false; error: E };
+
+const signer = getSigner(CHAIN);
+const encryptedHeader = await getAuthorizationHeader(signer);
+const publicClient = getPublicClient(CHAIN);
+
+export const merge = async (
+  maciContractAddress: Address,
+  pollId: number,
+  approval: string,
+  sessionKeyAddress: Address
+): Promise<CoordinatorServiceResult<boolean>> => {
```

Comment:
Use one object instead of four separate params

Reply:
These args have changed now but have done this where multiple args still exist

Reply:
So this is the idea that if I want to add new argument, I just add a new property to the object, plus I don't need to worry about the order of the arguments. For instance first option looks better than second one:
```
merge({ maciContractAddress: "0x...", pollId: 0n, approval: "321...", sessionKeyAddress: "123..." });
merge("0x...", 0n,  "321...",  "123...");
```

Reply:
Got it. Yeah I agree named arguments are cleaner - thanks for pointing this out!